matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+
+    public static final LocalDate DEFAULTDATE = LocalDate.of(2020, Month.JANUARY, 1);
+
+    private static final Parameter IGNORE_BASE_VOLTAGE_PARAMETER = new Parameter("ignore-base-voltage",
+            ParameterType.BOOLEAN,
+            "Ignore base voltage specified in the file",
+            Boolean.TRUE);
+
+    private static final class PerUnitContext {
+
+        private final double baseMva; // base apparent power
+
+        private final boolean ignoreBaseMva;
+
+        private PerUnitContext(double baseMva, boolean ignoreBaseMva) {
+            this.baseMva = baseMva;
+            this.ignoreBaseMva = ignoreBaseMva;
+        }
+
+        public boolean isIgnoreBaseMva() {
+            return ignoreBaseMva;
+        }
+
+        private double getBaseMva() {
+            return baseMva;
+        }
+    }
+
+    private static ContainersMapping createContainerMapping(MatpowerModel matpowerModel) {
+        ContainersMapping containersMapping = new ContainersMapping();
+
+        // group buses connected to non impedant lines to voltage levels
+        createVoltageLevelMapping(matpowerModel, containersMapping);
+
+        // group voltage levels connected by transformers to substations
+        createSubstationMapping(matpowerModel, containersMapping);
+
+        return containersMapping;
+    }
+
+    private static boolean isTransformer(MBranch branch) {
+        return branch.getRatio() != 0;
+    }
+
+    private static void createSubstationMapping(MatpowerModel model, ContainersMapping containersMapping) {
+        UndirectedGraph<String, Object> sGraph = new Pseudograph<>(Object.class);
+        for (String voltageLevelId : containersMapping.voltageLevelIdToBusNums.keySet()) {
+            sGraph.addVertex(voltageLevelId);
+        }
+        for (MBranch branch : model.getBranches()) {
+            if (isTransformer(branch)) {
+                sGraph.addEdge(containersMapping.busNumToVoltageLevelId.get(branch.getFrom()),
+                        containersMapping.busNumToVoltageLevelId.get(branch.getTo()));
+            }
+        }
+        int substationNum = 1;
+        for (Set<String> voltageLevelIds : new ConnectivityInspector<>(sGraph).connectedSets()) {
+            String substationId = "S" + substationNum++;
+            for (String voltageLevelId : voltageLevelIds) {
+                containersMapping.voltageLevelIdToSubstationId.put(voltageLevelId, substationId);
+            }
+        }
+    }
+
+    private static void createVoltageLevelMapping(MatpowerModel model, ContainersMapping containersMapping) {
+        UndirectedGraph<Integer, Object> vlGraph = new Pseudograph<>(Object.class);
+        for (MBus mBus : model.getBuses()) {
+            vlGraph.addVertex(mBus.getNumber());
+        }
+        for (MBranch mBranch : model.getBranches()) {
+            if (mBranch.getR() == 0 && mBranch.getX() == 0) {
+                vlGraph.addEdge(mBranch.getFrom(), mBranch.getTo());
+            }
+        }
```

Comment:
To be check, regarding my previous comment

Reply:
fixed